simple_graph.py

```diff
@@ -27,9 +28,20 @@ def add_edge(self, node1, node2):
             self.graph.setdefault(node1, [node2])
 
     def del_node(self, node):
-        self.graph.pop(node, None)
+        """Delete a node from the graph. Throw a KeyError if the node wasn't
+        in the graph."""
+        try:
+            self.graph.pop(node)
+        except KeyError:
+            raise KeyError("{} does not exist".format(node))
```

Comment:
I've commented on a couple of other people's submissions about this.  I think mutating the existing `KeyError` would be better than raising an entirely new one, as the location of the error will now be wrong.